UserDatabase.cpp

```diff
@@ -47,9 +47,9 @@ User UserDatabase::getCurrentUser() {
     return this->currentUser;
 }
 
-bool UserDatabase::searchUser(string username) {
-	bool userFound = false;
+User UserDatabase::searchUser(string username) {
     const char* dbName = "../users.db";
+    User user;
 
     sqlite3 *usersDB;
     string findQuery = "SELECT * FROM USERS where username = '" + username + "';";
```

Comment:
At the end of the statement '" + username "' can be replaced with ?. The bind text statement will replace ? with the username you are searching.

Reply:
fixed it